tutorial/tutorial.md

```diff
@@ -0,0 +1,333 @@
+
+To create the cluster, checkout the fleet-shard repository and login inside 
+
+`cd cos-fleet-shard`
+
+Create a minikube local install. This will create a local minikube profile called `cos`
+
+```shell
+./etc/scripts/start_minikube.sh
+```
+
+We can check the profile is installed correctly with 
+
+```
+minikube profile list
+```
+
+Install with this command all the CRD and the services needed by RHOC
+
+```
+kubectl apply -k etc/kubernetes/manifests/overlays/local --server-side --force-conflicts
+kubectl apply -f etc/kubernetes/manifests/overlays/local/camel-k/integration-platform.yaml --server-side  --force-conflicts
+```
+
+To verify everything was installed correctly, you can run
+
+`kubectl get namespaces`
+
+and verify the `redhat-openshift-connectors` is created.
+
+Inside this namespace, `camel-k` and `strimzi` will be running, you can check them with:
+
+`kubectl -n redhat-openshift-connectors get pods`
+
+After having created the minikube install, we should register it inside the control plane.
+Move to the `cos-tools` tab
+
+```
+export SUFFIX=$(uuidgen | tr -d '-')
+bin/create-cluster "$USER-$SUFFIX"
+```
+
+This will return something similar
+
+```json
+{
+  "id": "cg10fpruns2rbjn4ka20",
+  "kind": "ConnectorCluster",
+  "href": "/api/connector_mgmt/v1/kafka_connector_clusters/cg10fpruns2rbjn4ka20",
+  "owner": "lmolteni",
+  "created_at": "2023-03-03T15:44:56.505522+01:00",
+  "modified_at": "2023-03-03T15:44:56.505522+01:00",
+  "name": "lmolteni-6225FC431494432",
+  "annotations": {
+    "cos.bf2.org/organisation-id": "432432"
+  },
+  "status": {
+    "state": "disconnected"
+  }
+}
+```
+
+
+The creation of the cluster also creates a service account with the name of the cluster, if you want to check the created service-account you can use `rhoas`
+
+`rhoas service-account list`
+
+The cluster is in `disconnected` state. This means the control plane and the cluster aren't actively communicating.
+We'll fix it soon, but first let's generate a few secrets and insert them in the local cluster. 
+
+Gather the id from the previous request (in this case `cg10fpruns2rbjn4ka20`) and run
+
+```
+kubectl config set-context --current --namespace=redhat-openshift-connectors
+
+bin/create-cluster-secret <CLUSTER_ID>`
+```
+
+The first line is needed as `create-cluster-secret` needs to know the namespace where to create the secret.
+In this secret there will be a few parameters such as the `clientId` and the `clientSecret` of the service account user
+created during the cluster creation. Such parameters will be used by the `cos-fleetshard-sync` to communicate with the control plane. 
+
+To verify the secret was created accordingly, run
+
+`kubectl -n redhat-openshift-connectors get secret -o yaml`
+
+In another tab, run the `cos-fleet-shard-sync` module.
+
+```
+cd cos-fleetshard/cos-fleetshard-sync
+mvn -Dlocal -Dcontrol-plane-base-url=http://localhost:8000
```

Comment:
`-Dcontrol-plane-base-url` is likely not needed as the control plane should provide the right url

Reply:
@lucamolteni wrote about the need to override it "...but due to a bug it's being set to `localhost` instead..." below.